topdown/eval.go

```diff
@@ -2791,6 +2795,10 @@ func (e evalVirtualComplete) evalValue(iter unifyIterator, findOne bool) error {
 		return err
 	}
 
+	if prev == nil {
```

Comment:
I was wondering what happened with early-exit -- if rule bodies were amendable to the EE optimization, this would never be reached in the EE case -- **but** that doesn't matter, because this is about saving _undefined_. ✅